### PR TITLE
Implemented download links

### LINKED ISSUE #1863: Add downloadLink to File type in graphQL
Add `downloadLink` and populate with a download URL from the S3 bucket.

See #1695 for complete explanation. This has been split out to aid flow.

### PR DESCRIPTION
**Pull request by erezmus**
#### Background

* added downloadLink to GraphQL schema
* added getDownloadLink method to s3Storage
* new find method for Manuscript use case, updated resolver
* added test to Manuscript use case find method

#### Any relevant tickets

Closes #1863 

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

### CODE REVIEW

**Review by diversemix — APPROVED** [review]
Looking good!

**Review by will-byrne — APPROVED** [review]

File: packages/component-submission/server/resolvers/uploadSupportingFile.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 async function uploadSupportingFile(_, { file, id }, { user }) {
   const userUuid = await User.getUuidForProfile(user)
   const files = new SupportingFiles(S3Storage, id, userUuid)
   await files.upload(file)
-  const manuscript = await Manuscript.find(id, userUuid)
-  return manuscript
+
+  const manuscript = new Manuscript(config, userUuid, S3Storage)
+
+  return manuscript.getView(id)
```
Comment (diversemix) [inline packages/component-submission/server/resolvers/uploadSupportingFile.js:13]: does this need an `await` ?

File: packages/component-submission/server/resolvers/uploadSupportingFile.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+const { SupportingFiles, Manuscript } = require('../use-cases')
 
 async function uploadSupportingFile(_, { file, id }, { user }) {
   const userUuid = await User.getUuidForProfile(user)
   const files = new SupportingFiles(S3Storage, id, userUuid)
   await files.upload(file)
-  const manuscript = await Manuscript.find(id, userUuid)
-  return manuscript
+
+  const manuscript = new Manuscript(config, userUuid, S3Storage)
```
Comment (Cooryd) [inline packages/component-submission/server/resolvers/uploadSupportingFile.js:11]: I think this needs to be `await Manuscript.find(id, userUuid)` ?

Comment (diversemix) [reply packages/component-submission/server/resolvers/uploadSupportingFile.js:11]: I believe the `manuscript.find` happens inside `getView`

File: packages/component-submission/server/resolvers/uploadSupportingFile.js (hunk 1 above)
Comment (erezmus) [reply packages/component-submission/server/resolvers/uploadSupportingFile.js:13]: async functions wrap result in a Promise anyway, so i don't think so....

File: packages/component-submission/server/resolvers/uploadSupportingFile.js (hunk 2 above)
Comment (erezmus) [reply packages/component-submission/server/resolvers/uploadSupportingFile.js:11]: that's correct, its now the use case object which handles that part

**Review by Cooryd — APPROVED** [review]

**Pull request merged**